### PR TITLE
[BEAM-5730] Migrate ITs using DataflowRunner to use custom worker

### PR DESCRIPTION
**Pull request by boyuanzz**
Commit https://github.com/apache/beam/commit/e1b7a9be3a4a73460e1f4cdacf839638e2b2d889 migrated java dataflow runner ITs with legacy java worker to use custom worker jar. R: @herohde, @lukecwik 

Commit https://github.com/apache/beam/commit/3eba07962939486b83fbec67352e5cfc15c0e5ee changed the way setting up experiments. We shouldn't use Immutable list to set. R: @pabloem

### CODE REVIEW

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
All tests passed. Please review this PR @lukecwik

File: runners/google-cloud-dataflow-java/build.gradle [hunk 1]
```diff
@@ -104,12 +105,18 @@ test {
...
   def dataflowProject = project.findProperty('dataflowProject') ?: 'apache-beam-testing'
   def dataflowTempRoot = project.findProperty('dataflowTempRoot') ?: 'gs://temp-storage-for-validates-runner-tests/'
+  def dataflowWorkerJar = project.findProperty('dataflowWorkerJar') ?: project(":beam-runners-google-cloud-dataflow-java-legacy-worker").shadowJar.archivePath
+
   systemProperty "beamTestPipelineOptions", JsonOutput.toJson([
           "--runner=TestDataflowRunner",
           "--project=${dataflowProject}",
           "--tempRoot=${dataflowTempRoot}",
+          "--dataflowWorkerJar=${dataflowWorkerJar}",
+          "--workerHarnessContainerImage=",
```
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/build.gradle]: Why do we leave a blank workerHarnessContainerImage?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java [hunk 2]
```diff
@@ -291,8 +292,10 @@ public void testStandardQueryWithoutCustom() throws Exception {
   @Test
   public void testNewTypesQueryWithoutReshuffleWithCustom() throws Exception {
     this.setupNewTypesQueryTest();
-    this.options.setExperiments(
-        ImmutableList.of("enable_custom_bigquery_sink", "enable_custom_bigquery_source"));
+    List<String> experiments = new ArrayList<>();
+    experiments.add("enable_custom_bigquery_sink");
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: Why did you need to make this change?

File: runners/google-cloud-dataflow-java/build.gradle (hunk 1 above)
Comment (herohde) [reply runners/google-cloud-dataflow-java/build.gradle]: To pick up the non-versioned container image, which handles a staged worker jar.

Comment (boyuanzz) [reply runners/google-cloud-dataflow-java/build.gradle]: This is only for the legacy worker and related to the Dataflow service. If the workerHarnessContainerImage isn't set to empty explicit, it will use the default one which contains the pre-build worker jar.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java (hunk 2 above)
Comment (boyuanzz) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: If there is dataflowWorkerJar set, then we will append the experiment flag into experiments: https://github.com/apache/beam/blob/master/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L744. In this case, if the experiments from pipeline is a immutable collection, then when trying to append the worker_jar experiment, an exception will be thrown.

Comment (lukecwik) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: Wouldn't it make sense to fix the underlying bug to correctly concatenate two lists together instead of assuming the underlying list is mutable?

File: runners/google-cloud-dataflow-java/build.gradle (hunk 1 above)
Comment (lukecwik) [reply runners/google-cloud-dataflow-java/build.gradle]: Seems worthwhile for a comment since all other properties make sense since we are setting an explicit value.

**Comment by boyuanzz** [comment]
This PR is ready to merge if there is no more concerns. @lukecwik

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
R: @lukecwik , latest commits address comments above.
